parser/test/astnodes/SwitchStmtTest.c

```diff
@@ -0,0 +1,41 @@
+#include <stdio.h>
+#include <stdbool.h>
+#include <assert.h>
+
+#include "SwitchStmtTest.h"
+#include "../../main/commandline/TokenList.h"
+#include "../../main/commandline/TokenKeys.h"
+#include "../../main/parsing/statements/SwitchStmt.h"
+#include "../../../token/token.h"
+#include "../../../ast/free_ast.h"
+
+int test_switch(bool debug){
+
+	if(debug){
+		printf("TEST: test_switch()\n");
+	}
+
+	struct TokenList* list = makeTokenList();
+	
+	list_add(list, makeToken2(SWITCH,"switch"));
+	list_add(list, makeToken2(ID,"a"));
+	
+	list_add(list, makeToken2(LCURLY,"{"));
+	
+	list_add(list, makeToken2(CASE,"case"));
+	list_add(list, makeToken2(INTEGER,"4"));
+	
+	list_add(list, makeToken2(LCURLY,"{"));
+	list_add(list, makeToken2(RCURLY,"}"));
+	
+	list_add(list, makeToken2(RCURLY,"}"));
+	
+	struct SwitchStmt* t = makeSwitchStmt(list,debug);
+	
+	assert(t != NULL);
```

Comment:
there should be more assertions, to look if the SwitchStmt has been correctly initialized

Reply:
resolved